allennlp/training/trainer.py

```diff
@@ -240,7 +247,12 @@ def __init__(self,
         self._train_data = train_dataset
         self._validation_data = validation_dataset
 
-        self._patience = patience
+        if patience == 0:
```

Comment:
Looks great overall, but I think it would be an improvement to require `patience > 0` and accept only `None` to disable it.  Otherwise we'll get the question "What is `patience == -2"!

Reply:
@schmmd I don't think that's a problem. From the doc it's pretty clear what happens for negative patience.